module/zfs/dmu_objset.c

```diff
@@ -952,7 +969,7 @@ dmu_objset_evict(objset_t *os)
 		mutex_exit(&os->os_lock);
 	}
 
-
+	(void) atomic_cas_64(&os->os_ev_tid, tid, 0);
```

Comment:
This needs to do a `cv_broadcast()` to ensure that it wakes up any threads that may be waiting on the cv. This should be done under the mutex, which eliminates the need for the atomic cas operation.